### PR TITLE
[PROD-1032] Actually delete the cluster

### PR DESCRIPTION
**Pull request by rmoneys**
https://synccomputing.atlassian.net/browse/PROD-1032

Make a best effort to delete clusters created by the library.

### CODE REVIEW

File: sync/cli/awsdatabricks.py [hunk 1]
```diff
@@ -74,16 +76,20 @@ def run_job(
     default=False,
     help="Force creation of a prediction even with incomplete cluster data.",
 )
+@click.option(
+    "--exclude-task", help="Don't consider task when finding the cluster of a run", multiple=True
```
Comment (rmoneys) [inline sync/cli/awsdatabricks.py:80]: Pete mentioned this might be handy

File: sync/awsdatabricks.py [hunk 2]
```diff
@@ -659,19 +659,20 @@ def get_project_cluster_settings(project_id: str, region_name: str = None) -> Re
...
-def run_job_object(job: dict) -> Response[str]:
+def run_job_object(job: dict) -> Response[Tuple[str, str]]:
     """Create a Databricks one-off run based on the job configuration.
 
     :param job: Databricks job object
     :type job: dict
-    :return: run ID
-    :rtype: Response[str]
+    :return: run ID, and optionally ID of newly created cluster
+    :rtype: Response[Typle[str, str]]
```
Comment (gseyffert-sync) [inline sync/awsdatabricks.py:668]: nit - `Tuple`

File: sync/awsdatabricks.py [hunk 3]
```diff
@@ -1003,7 +1010,7 @@ def terminate_cluster(cluster_id: str) -> Response[dict]:
         elif state == "TERMINATING":
             return _wait_for_cluster_termination(cluster_id)
         elif state in {"PENDING", "RUNNING", "RESTARTING", "RESIZING"}:
-            get_default_client().delete_cluster(cluster_id)
+            get_default_client().terminate_cluster(cluster_id)
```
Comment (gseyffert-sync) [inline sync/awsdatabricks.py:1013]: makes sense

**Review by gseyffert-sync — CHANGES_REQUESTED** [review]
lgtm, but the typo in the docs should probably be fixed first.

File: sync/awsdatabricks.py (hunk 2 above)
Comment (rmoneys) [reply sync/awsdatabricks.py:668]: I wish it was "Typle"

**Review by gseyffert-sync — APPROVED** [review]

File: sync/api/projects.py [hunk 4]
```diff
@@ -58,7 +58,7 @@ def create_project(
     :type s3_url: str, optional
     :param prediction_preference: preferred prediction solution, defaults to `Preference.ECONOMY`
     :type prediction_preference: Preference, optional
-    :param prediction_params: dictionary of prediction parameters, defaults to None. Valid options are documented here - https://developers.synccomputing.com/reference/create_project_v1_projects_post
+    :param prediction_params: dictionary of prediction parameters, defaults to None. Valid options are documented `here <https://developers.synccomputing.com/reference/create_project_v1_projects_post>`__
```
Comment (rmoneys) [inline sync/api/projects.py:61]: Actually resolve the presentation issue noted here: https://github.com/synccomputingcode/syncsparkpy/pull/30#issuecomment-1603270800

**Pull request merged**